specs/MultiProfile.md

```diff
@@ -0,0 +1,146 @@
+API spec for multiple profile support
+===
+
+# Background
+
+Currently, all WebView2s can only use one fixed Edge profile in the browser process, which is
+normally the **Default** profile or the profile specified by the **--profile-directory** command
+line switch. Regarding this we have got a bunch of requests to support multiple profiles, so we're
+adding these new APIs.
+
+With this you can have different WebView2s running with separate profiles under a given user data
+directory, which means separate cookies, user preference settings, and various data storage etc.,
+to enable you to build a more wonderful experience for your application.
+
+# Examples
+<!-- TEMPLATE
+    Use this section to explain the features of the API, showing
+    example code with each description in both C# (for our WinRT API or .NET API) and
+    in C++ for our COM API. Use snippets of the sample code you wrote for the sample apps.
+    The sample code for C++ and C# should demonstrate the same thing.
+
+    The general format is:
+
+    ## FirstFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    ## SecondFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    As an example of this section, see the Examples section for the Custom Downloads
+    APIs (https://github.com/MicrosoftEdge/WebView2Feedback/blob/master/specs/CustomDownload.md). 
+-->
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2Staging6;
+interface ICoreWebView2StagingCreateControllerOptions;
+interface ICoreWebView2StagingEnvironment4;
+interface ICoreWebView2StagingProfile;
+
+[uuid(57FD205C-39D5-4BA1-8E7B-3E53C323EA87), object, pointer_default(unique)]
+interface ICoreWebView2StagingEnvironment4 : IUnknown
+{
+  /// Create a new ICoreWebView2StagingCreateControllerOptions to be passed as a parameter of
+  /// CreateCoreWebView2ControllerWithOptions and CreateCoreWebView2CompositionControllerWithOptions.
+  HRESULT CreateCoreWebView2CreateControllerOptions(
```

Comment:
Maybe this should just be ControllerOptions instead of CreateControllerOptions (and also ICoreWebView2StagingCreateControllerOptions -> ICoreWebView2StagingControllerOptions). What do you think?
```suggestion
  HRESULT CreateCoreWebView2ControllerOptions(
```

Reply:
Done. I also think CreateCoreWebView2ControllerOptions is more concise. Later will update our code base accordingly.